airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -266,21 +266,26 @@ public AutoCloseableIterator<JsonNode> queryTableIncremental(final JdbcDatabase
                                                                final String tableName,
                                                                final String cursorField,
                                                                final Datatype cursorFieldType,
-                                                               final String cursor) {
+                                                               final String cursorValue) {
     LOGGER.info("Queueing query for table: {}", tableName);
     return AutoCloseableIterators.lazyIterator(() -> {
       try {
         final Stream<JsonNode> stream = database.unsafeQuery(
             connection -> {
               LOGGER.info("Preparing query for table: {}", tableName);
-              final String sql = String.format("SELECT %s FROM %s WHERE %s > ?",
+              final String quotedCursorField = sourceOperations.enquoteIdentifier(connection, cursorField);
+              final StringBuilder sql = new StringBuilder(String.format("SELECT %s FROM %s WHERE %s > ?",
                   sourceOperations.enquoteIdentifierList(connection, columnNames),
-                  sourceOperations
-                      .getFullyQualifiedTableNameWithQuoting(connection, schemaName, tableName),
-                  sourceOperations.enquoteIdentifier(connection, cursorField));
+                  sourceOperations.getFullyQualifiedTableNameWithQuoting(connection, schemaName, tableName),
+                  quotedCursorField));
+              // if the connector emits intermediate states, the incremental query must be sorted by the cursor
+              // field
+              if (getStateEmissionFrequency() > 0) {
+                sql.append(String.format(" ORDER BY %s ASC", quotedCursorField));
```

Comment:
Probably a dumb question but are we sure `ORDER BY %s ASC` would be compatible with all the AbstractJDBC based sources? I assume it should be!

Reply:
Good point. This is basic ANSI SQL syntax, so I think all of them should follow it. But I will double check.